app/Http/UseCase/KeyResult/GetIndexData.php

```diff
@@ -43,9 +43,16 @@ public function __invoke(array $input): array
         $isArchive = $input['is_archive'] ?? null;
         $isArchive = is_null($isArchive) ? false : (bool) $isArchive;
 
+        $comments = $this->commentRepo->getByObjectiveId($objectiveId);
+
+        //コメントのいいね情報を取得して$commentsに統合
+        foreach ($comments as $key => $comment) {
+            $comments[$key]->isLiked = $this->commentLikeUserRepo->isLikedBy($comment->id, auth()->id());
```

Comment:
https://github.com/EeeeG-Inc/OKR-web-app/blob/feature/comment-make-like-button/app/Models/Comment.php#L68

本来 Comment モデルには liceCount や isLiked といった attributes は存在してないですし、
comments テーブルにそういったカラムも存在していなかったと思います。

つまりここでやっているのは、この場限定で Comment モデルにグローバルな変数を持たせているということです。
動いているとは思いますが、
この場限定の処理なので、他のエンジニアは勘違いを生みやすいかと思います。

View で `$comment->isLiked` してたり、
Controller で dd したら `$comment->isLiked` なるものが見えるので、
なるほど、このモデルはそういう attribute が備わっているのだな、という勘違いですね

個人的には少し気持ち悪い実装なので、
モデルに getAttribure メソッドを実装して存在を明示的にしてもらうか、
まったく別の変数を用意して View に渡すか、がいいと思います ※ わざわざ Comment モデルに値を設定している理由もないと思いますので

Reply:
```php
foreach ($comments as $key => $comment) {
            $comments[$key]->isLiked 
```

ちなみにここですが、
せっかく `$comment` で要素を取り出しているのだから
わざわざ  `$comments[$key]` で指定する必要ないかと思います。

```suggestion
        foreach ($comments as $comment) {
            $comment->isLiked = $this->commentLikeUserRepo->isLikedBy($comment->id, auth()->id());
```

この処理だったら `$key` 自体がそもそも不要ですね